### PR TITLE
move function GetKubemarkMasterComponentResoureUsage and remove long-time TODO

### PR DESCRIPTION
**Pull request by tanjunchen**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

ref:https://github.com/kubernetes/kubernetes/issues/86763

**Special notes for your reviewer**:
see https://github.com/kubernetes/kubernetes/pull/87271#discussion_r367215524
/cc @neolit123 
/cc @oomichi 

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

File: test/e2e/framework/resource_usage_gatherer.go [hunk 1]
```diff
@@ -556,3 +557,65 @@ func (g *ContainerResourceGatherer) StopAndSummarize(percentiles []int, constrai
...
+
+func getMasterUsageByPrefix(prefix string) (string, error) {
+	sshResult, err := e2essh.SSH(fmt.Sprintf("ps ax -o %%cpu,rss,command | tail -n +2 | grep %v | sed 's/\\s+/ /g'", prefix), GetMasterHost()+":22", TestContext.Provider)
+	if err != nil {
+		return "", err
+	}
+	return sshResult.Stdout, nil
+}
+
+// getKubemarkMasterComponentsResourceUsage returns the resource usage of kubemark which contains multiple combinations of cpu and memory usage for each pod name.
```
Comment (tanjunchen) [inline test/e2e/framework/resource_usage_gatherer.go]: see https://github.com/kubernetes/kubernetes/pull/87271/files#diff-b498843cdcc70479a96c0cfffc6290afL44

This TODO from https://github.com/kubernetes/kubernetes/pull/38507/files#diff-b498843cdcc70479a96c0cfffc6290afR31 4 years ago.
GetKubemarkMasterComponentsResourceUsage  is  only used by https://github.com/kubernetes/kubernetes/pull/87271/files#diff-572985ec546140c5cf2bc24658132e83R187
so make it private.

For 
```
TODO: figure out how to move this to kubemark directory (need to factor test SSH out of e2e framework
```
I think that it is good time to re-examine and evaluate this TODO

if someone wishes to write a separate proposal for this work, please create an issue or TODO(@github) named to this function, We can easily track this down

File: test/e2e/framework/resource_usage_gatherer.go [hunk 2]
```diff
@@ -556,3 +557,65 @@ func (g *ContainerResourceGatherer) StopAndSummarize(percentiles []int, constrai
 	}
 	return &summary, nil
 }
+
+// KubemarkResourceUsage is a struct for tracking the resource usage of kubemark.
+type KubemarkResourceUsage struct {
```
Comment (neolit123) [inline test/e2e/framework/resource_usage_gatherer.go]: can this structure become private

`kubemarkResourceUsage`?

Comment (tanjunchen) [reply test/e2e/framework/resource_usage_gatherer.go]: yes . sorry forget it

File: test/e2e/framework/resource_usage_gatherer.go [hunk 3]
```diff
@@ -183,7 +184,7 @@ type resourceGatherWorker struct {
 func (w *resourceGatherWorker) singleProbe() {
 	data := make(ResourceUsagePerContainer)
 	if w.inKubemark {
-		kubemarkData := GetKubemarkMasterComponentsResourceUsage()
+		kubemarkData := getKubemarkMasterComponentsResourceUsage()
```
[the remark this answers is not in the document]
Comment (tanjunchen) [reply test/e2e/framework/resource_usage_gatherer.go]: sure. 
```
TODO: move this to test/e2e/scalability see https://github.com/kubernetes/kubernetes/pull/87271#discussion_r367217336
```
We can explain this here.How about it?

[the remark this answers is not in the document]
Comment (neolit123) [reply test/e2e/framework/resource_usage_gatherer.go]: lets not add the TODO just yet.

i would like to see what others think about the new proposed location and decoupling the framework from kubemark.

[the remark this answers is not in the document]
Comment (tanjunchen) [reply test/e2e/framework/resource_usage_gatherer.go]: ok

**Pull request merged**